### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by astanishevskyi**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import (
+	"github.com/astanishevskyi/http-server/apiserver"
+	"github.com/astanishevskyi/http-server/apiserver/configs"
+	"log"
+)
+
+func main() {
+	config := configs.Config{
+		BindAddr: ":8080",
+		Storage:  "in-memory",
```
Comment (dmtbrk) [inline main.go]: This type of storage configuration seems to be easier for users but doesn't allow them to write their own implementation. Consider if it is what you want.

File: apiserver/models/user.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package models
+
+type User struct {
+	Id    uint32 `json:"id"`
+	Name  string `json:"name"`
+	Email string `json:"email"`
+	Age   uint8  `json:"age"`
+}
+
+type UserInterface interface {
```
Comment (dmtbrk) [inline apiserver/models/user.go]: `UserService` or `UserStorage` is a common name for this kind of interface.

File: apiserver/models/user.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package models
+
+type User struct {
+	Id    uint32 `json:"id"`
```
Comment (dmtbrk) [inline apiserver/models/user.go]: `ID` is preffered. Here's why: https://github.com/golang/go/wiki/CodeReviewComments#initialisms

File: apiserver/server.go [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+			_, err = w.Write(respJson)
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+		}
+	case http.MethodPost:
+		log.Println("POST /user/")
+		if err := r.ParseMultipartForm(10000); err != nil {
+			log.Fatal(err)
```
Comment (dmtbrk) [inline apiserver/server.go]: This will exit the program. Do you really want it if you've got just a malformed request?

File: apiserver/server.go [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+	http.HandleFunc("/user/", s.User)
+}
+
+func (s *Server) User(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case http.MethodGet:
+
+		id, _ := strconv.ParseUint(strings.TrimPrefix(r.URL.Path, "/user/"), 0, 32)
+
+		if id == 0 {
```
Comment (dmtbrk) [inline apiserver/server.go]: If you don't want to return all users on requests like `/user/gibberish`, use some other logic. It is expected to return an error (400, 404, or other) instead.

File: apiserver/adapters/in_memory_adapter.go [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+func (s *UserStorage) GetAll() []models.User {
+	return s.UsersSlice
+}
+
+func (s *UserStorage) Retrieve(id uint32) (models.User, error) {
+	for _, val := range s.UsersSlice {
+		if val.Id == id {
+			return val, nil
+		}
+	}
```
Comment (dmtbrk) [inline apiserver/adapters/in_memory_adapter.go]: What is the time complexity of this algorithm and what can be done to improve it?

File: apiserver/adapters/in_memory_adapter.go [hunk 7]
```diff
@@ -0,0 +1,59 @@
+package adapters
+
+import (
+	"errors"
+	"github.com/astanishevskyi/http-server/apiserver/models"
+	"math/rand"
+)
+
+type UserStorage struct {
+	UsersSlice []models.User `json:" users"`
```
Comment (dmtbrk) [inline apiserver/adapters/in_memory_adapter.go]: Concurrent writes to this slice will be unsafe without using synchronization technics. Think why.

**Review by dmtbrk — COMMENTED** [review]
Nice

File: main.go (hunk 1 above)
Comment (astanishevskyi) [reply main.go]: Fixed. I have replaced this block with .env file

File: apiserver/models/user.go (hunk 3 above)
Comment (astanishevskyi) [reply apiserver/models/user.go]: Thanks, fixed

File: apiserver/server.go (hunk 4 above)
Comment (astanishevskyi) [reply apiserver/server.go]: Fixed

File: apiserver/models/user.go (hunk 2 above)
Comment (astanishevskyi) [reply apiserver/models/user.go]: Fixed

File: apiserver/adapters/in_memory_adapter.go (hunk 7 above)
Comment (astanishevskyi) [reply apiserver/adapters/in_memory_adapter.go]: Fixed

File: apiserver/server.go (hunk 5 above)
Comment (astanishevskyi) [reply apiserver/server.go]: Fixed

File: apiserver/adapters/in_memory_adapter.go (hunk 6 above)
Comment (astanishevskyi) [reply apiserver/adapters/in_memory_adapter.go]: This algorithm has O(n) complexity. I have no idea how to simplify it, please help

**Pull request merged**